emitter/test/emitter-test-part-1.js

```diff
@@ -3,129 +3,101 @@ const EventEmitter = require('../emitter-part-1.js');
+    it("can trigger an event", function (done) {   
+        emitter.on("Hello", () => done());
         emitter.trigger("Hello");
     });
-    it("can trigger an event with data", function (done) {
-        emitter.on("Hello", function (data) {
+    it("can trigger an event with data", function () {
+        emitter.on("Hello", (data) => {
             assert.equal(data, "World");
-            done();
         });
         emitter.trigger("Hello", "World");
 
     });
-    it("only triggers events once", function (done) {
-        let triggered = false;
-        emitter.on("Hello", function () {
-            if (triggered) throw new Error("Multiple triggers");
-            triggered = true;
-            setTimeout(done, 1);
-        });
+    it("Can trigger a event with no handlers", function () {
         emitter.trigger("Hello");
     });
+    it("only triggers events once", function () {
+        const trackedFunction = tracker.calls(1);
 
-    it("supports several event handlers", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("Hello", byOneTwo);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
-        
-        function byOneTwo() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
+        emitter.on("Hello", trackedFunction);
+        emitter.trigger("Hello");
+
+        tracker.verify();
+    });
+
+    it("supports several event handlers", function () {
+        const trackedFunction = tracker.calls(1);
+        const trackedFunction1 = tracker.calls(1);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("Hello", trackedFunction1);
         emitter.trigger("Hello");
+
+        tracker.verify();
     });
     
-    it("supports same function multiple times", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("Hello", byOne);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
-        
+    it("supports same function multiple times", function () {
+        const trackedFunction = tracker.calls(2);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("Hello", trackedFunction);
         emitter.trigger("Hello");
+
+        tracker.verify();
     });
 
-    it("Supports multiple event triggers same function", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("World", byOne);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
+    it("Supports multiple event triggers same function", function () {
+        const trackedFunction = tracker.calls(2);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction);
```

Comment:
Now we will need to explain the magic that it does behind the scenes instead of letting the interviewees see the code.

Reply:
3. there is a polyfill that does basically what was there before, so it can be read as well